File: app/src/main/java/com/example/muhammadjon/myproject/utils/NetUtils.java
package com.example.muhammadjon.myproject.utils;


import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class NetUtils {
    private ConnectivityManager manager;
    private static NetUtils INSTAINS;

    private NetUtils(Context ctx) {
        manager= (ConnectivityManager) ctx.getSystemService(Context.CONNECTIVITY_SERVICE);
    }

    public static NetUtils getINSTAINS(Context ctx) {
        if (INSTAINS==null) {
            INSTAINS=new NetUtils(ctx);
        }
        return INSTAINS;
    }

    public boolean isOnline() {
        NetworkInfo info = manager.getActiveNetworkInfo();
        return info != null && info.isConnectedOrConnecting();
    }
}

File: app/src/main/java/com/example/muhammadjon/myproject/utils/NetUtils.kt
package com.example.muhammadjon.myproject.utils


import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

class NetUtils private constructor(ctx: Context) {
    private val manager: ConnectivityManager = ctx.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager

    val isOnline: Boolean
        get() {
            val info = manager.activeNetworkInfo
            return info != null && info.isConnected
        }

    companion object {
        private var INSTAINS: NetUtils? = null

        fun getINSTAINS(ctx: Context): NetUtils {
            if (INSTAINS == null) {
                INSTAINS = NetUtils(ctx)
            }
            return INSTAINS as NetUtils
        }
    }
}
